common/kokkos-sampler/kp_sampler_skip.cpp

```diff
@@ -136,109 +139,203 @@ void kokkosp_init_library(const int loadSeq, const uint64_t interfaceVer,
 
   uniqID = 1;
 
-  const char* tool_sample = getenv("KOKKOS_TOOLS_SAMPLER_SKIP");
+  const char* tool_sample      = getenv("KOKKOS_TOOLS_SAMPLER_SKIP");
+  const char* tool_probability = getenv("KOKKOS_TOOLS_SAMPLER_PROBABILITY");
+  // composing the periodicity and probability parameter. Set to 1 if
+  // probability for each periodic sample. Set to 0 to default to probability
+  // even if periodicity (skip rate) is defined.
+
   if (NULL != tool_sample) {
     kernelSampleSkip = atoi(tool_sample) + 1;
   }
 
+  if (NULL != tool_probability) {
+    //  read sampling probability as an float between 0 and 100, representing
+    //  a percentage that data should be gathered.
+    //  Connector reasons about probability as a double between 0.0 and 1.0.
+    tool_prob_num = atof(tool_probability);
+    if (tool_prob_num > 100.0) {
+      printf(
+          "KokkosP: The sampling probability value is set to be greater than "
+          "100.0. "
+          "Setting sampling probability to 100 percent; all of the "
+          "invocations of a Kokkos Kernel will be profiled.\n");
+      tool_prob_num = 100.0;
+    } else if (tool_prob_num < 0.0) {
+      printf(
+          "KokkosP: The sampling probability value is set to be negative "
+          "number. Setting "
+          "sampling probability to 0 percent; none of the invocations of "
+          "a Kokkos Kernel will be profiled.\n");
+      tool_prob_num = 0.0;
+    }
```

Comment:
If both tool_prob_num < 0 and  kernelSampleSkip is max of uint64_t set tool_prob_num to 10.0

Make an error check checking that not. both of them are set

Reply:
The tool_prob_num has been updated as requested. 

 I have set kernelSampleSkip to be 0 instead of std::numeric_limits<uint64_t>::max due to the following error that I can't seem to figure out an immediate answer to: 

```
error: comparison between pointer and integer ('int64_t' (aka 'long long') and 'type (*)() noexcept' (aka 'unsigned long long (*)() noexcept'))
  if ((tool_prob_num < 0.0 ) && (kernelSampleSkip == std::numeric_limits<uint64_t>::max)) {
```

Reply:
The `tool_prob_num` default has been assigned to the default requested. The `kernelSampleSkip` default is part of a new PR which focuses just on the correct matching of sampled kernels.